src/nbl/system/ISystem.cpp

```diff
@@ -7,4 +7,46 @@ namespace nbl::system
     {
         addArchiveLoader(core::make_smart_refctd_ptr<CArchiveLoaderZip>());
     }
+    core::smart_refctd_ptr<IFile> ISystem::getFileFromArchive(const system::path& _path)
```

Comment:
shouldn't be a special separate method, but rolled into `ISystem::createFile` (open a "normal" file when a file archive can't be found and nested file cannot be opened)